File: result/translate/starchat/java2python/python/PRINT_MAXIMUM_SHORTEST_DISTANCE.py
def f_gold(a, n, k):
        b = dict()
        for i in range(n):
            x = a[i]
            d = min(1 + i, n - i)
            if x not in b:
                b[x] = d
            else:
                b[x] = min(d, b[x])
        ans = float('inf')
        for i in range(n):
            x = a[i]
            if x!= k - x and k - x in b:
                ans = min(max(b[x], b[k - x]), ans)
        return ans

File: result/translate/chatgpt/python2java/java/PRINT_MAXIMUM_SHORTEST_DISTANCE.java
public int f_gold(int[] a, int n, int k) {
    Map<Integer, Integer> b = new HashMap<>();
    for (int i = 0; i < n; i++) {
        int x = a[i];
        int d = Math.min(1 + i, n - i);
        if (!b.containsKey(x)) {
            b.put(x, d);
        } else {
            b.put(x, Math.min(d, b.get(x)));
        }
    }
    int ans = Integer.MAX_VALUE;
    for (int i = 0; i < n; i++) {
        int x = a[i];
        if (x != (k - x) && b.containsKey(k - x)) {
            ans = Math.min(Math.max(b.get(x), b.get(k - x)), ans);
        }
    }
    return ans;
}